### PR TITLE
Adds donation controller for Stripe checkout

### PR DESCRIPTION
**Pull request by ribordy**
This handles creating a Stripe Checkout session for a given donation amount. At present it's mostly an exploration of interacting with the Stripe API; I'd like to harden against bad inputs & add some tests before I feel good about merging this in.

### CODE REVIEW

File: src/controller/DonationsController.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        line_items: [{
+          description: `About ${numberOfPizzas} ${numberOfPizzas === 1 ? 'pizza' : 'pizzas'}`,
+          price_data: {
+            product: process.env.DONATION_PRODUCT_ID,
+            unit_amount: amountUsd * 100,
+            currency: 'usd',
+          },
+          quantity: 1,
+        }],
+        mode: 'payment',
```
Comment (mojowen) [inline src/controller/DonationsController.ts:27]: i think we'll also want to add `metadata` that accepts a `referrer` URL param from the frontend.

File: src/routes.ts [hunk 2]
```diff
@@ -103,4 +104,10 @@ export const Routes = [
     controller: RootController,
     action: "root",
   },
+  {
+    method: "post",
```
Comment (mojowen) [inline src/routes.ts:108]: 🙌

File: src/controller/DonationsController.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+export class DonationsController {
+  async create(request: Request, _response: Response, _next: NextFunction) {
+    const {
+      amountUsd,
+    } = request.body;
+
+    const numberOfPizzas = Math.ceil(amountUsd / 20);
+    try {
+      const stripe = new Stripe(process.env.STRIPE_SECRET_KEY, {
+        apiVersion: '2020-08-27',
```
Comment (mojowen) [inline src/controller/DonationsController.ts:13]: Any reason we're not using `2020-10-24` ?

**Review by mojowen — COMMENTED** [review]
Looks great! Some basic smoke tests would be useful - feel free to stub the stripe endpoint and in `tests/__mock__/stripe` or something. I'll build my branch off of this

File: src/controller/DonationsController.ts (hunk 3 above)
Comment (ribordy) [reply src/controller/DonationsController.ts:13]: Nope. It wasn't out when I did the initial exploratory work, and I hadn't checked to see if there was a new version yet. Will make the change!

Comment (mojowen) [reply src/controller/DonationsController.ts:13]: Oh I didn't check if it was out now so maybe this is a noop

Comment (ribordy) [reply src/controller/DonationsController.ts:59]: It appears that 2020-08-27 is the most recent version.

File: src/controller/DonationsController.ts [hunk 4]
```diff
@@ -1,15 +1,17 @@
 import { NextFunction, Request, Response } from "express";
-import * as Stripe from "stripe";
```
Comment (ribordy) [inline src/controller/DonationsController.ts:2]: @mojowen I made a few changes to your work in progress to get this up and running--let me know if you wanna run through the changes sometime this week!

**Review by ribordy — COMMENTED** [review]
I think this should be good to go now! Next item of work is retooling the front end to use this endpoint and accept the success callback params to show the post-donate UI.

**Review by mojowen — APPROVED** [review]

**Comment by ribordy** [comment]
Closing in favor of #24

**Pull request closed without merging**